File: frontend/src/types/photo.ts
/**
 * Photo types
 */

export interface Photo {
  id: string;
  plant_id: string;
  file_path: string;
  thumbnail_path: string | null;
  original_filename: string;
  file_size: number;
  mime_type: string;
  width: number | null;
  height: number | null;
  caption: string | null;
  taken_at: string | null;
  created_at: string;
}

export interface PhotoCreate {
  plant_id: string;
  caption?: string;
  taken_at?: string;
}

export interface PhotoUpdate {
  caption?: string;
  taken_at?: string;
}

File: backend/app/schemas/photo.py
"""Pydantic schemas for photo operations."""
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class PhotoBase(BaseModel):
    """Base photo schema."""

    caption: str | None = None
    taken_at: datetime | None = None


class PhotoCreate(PhotoBase):
    """Schema for creating a photo."""

    plant_id: UUID


class PhotoUpdate(BaseModel):
    """Schema for updating a photo."""

    caption: str | None = None
    taken_at: datetime | None = None


class PhotoResponse(PhotoBase):
    """Schema for photo response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    plant_id: UUID
    file_path: str
    thumbnail_path: str | None
    original_filename: str
    file_size: int
    mime_type: str
    width: int | None
    height: int | None
    created_at: datetime
